### PR TITLE
test: migrate lambda-form describe()/it() to @describe/@it (#1599)

### PR DESCRIPTION
**Pull request by t0k0sh1**
## Summary

- Migrate 167 of 170 `tests/spec/*.test.ry` files from the deprecated lambda form `describe("...", ():` / `it("...", ():` to the canonical `@describe("...")` / `@it("...")` named-function form
- Reduces `./build/ry test -p` deprecation warnings from 340 → 6
- Add `stripTestingImport()` helper to `tests/test_codegen_common.hpp` so the codegen test harness can still run migrated spec files (it skips ModuleLoader)
- Update `docs/reference/testing.md` canonical examples to the @directive form

## Background

PR #1424 made `@deprecated` warnings actually reach stderr. Prior to that, codegen's deprecation push for the lambda form (`src/codegen_test.cpp:57-58`, `:176-177`) was silently dropped because no production path called `getWarnings()`. After #1424 merged, every `./build/ry test -p` run emitted 340 warnings (170 files × 2 warnings each).

## What's NOT migrated

Three files remain in the deprecated lambda form because they expose a separate parser/codegen bug (filed as #1601):

- `tests/spec/numeric_literal_suffix.test.ry` — uses `f64` literal suffix
- `tests/spec/numeric_underscore_separator.test.ry` — uses `f64` literal suffix
- `tests/spec/operator_overload.test.ry` — uses local `record` declarations + custom `operator as`

Repro of #1601:

```ry
fn testF64():
  x = 3.14f64    # error: unknown type: f64

testF64()
```

The same `3.14f64` works at module top level and inside a lambda body, but fails in any named-function body. Lambda removal (#1602) is blocked on #1601 being fixed.

## Verification

- `./build/ry test -p` — 173 files, 0 failures, 6 deprecation warnings (down from 340)
- `./build/ry_tests` — 2124/2124 pass
- ASan+UBSan — all pass
- TSan — all pass
- libFuzzer — all corpus replay clean
- Absolute count check: 2757 `@it(` + 44 lambda `it(` + 12 (directive_test pre-existing) = 2789 ✓ matches baseline

## Test plan

- [x] `./build/ry test -p` passes with ≤6 deprecation warnings
- [x] `./build/ry_tests` all pass
- [x] ASan+UBSan / TSan / libFuzzer self-tests
- [x] Spot-check 10 random migrated files for naming consistency
- [x] `docs/reference/testing.md` examples render correctly

## Related

- Closes #1599
- Follow-up: #1601 (parser bug blocking remaining 3 files), #1602 (deprecated-form removal)

## Summary by CodeRabbit

* **Tests**
  * Migrated 167 test suites from the deprecated lambda-style to decorator-based @describe/@it, modernizing test registration and removing ~340 deprecation warnings; 3 remaining suites are pending due to a parser/codegen edge-case.

* **Documentation**
  * Updated testing reference to recommend the decorator-based function style, with lambda forms marked deprecated.

### CODE REVIEW

File: tests/spec/functions.test.ry [hunk 1]
```diff
@@ -34,73 +36,73 @@ fn operator==(a: Vec2, b: Vec2) -> bool:
...
-  )
-)
 
-describe("UFCS", ():
-  it("should support method-style call", ():
+@describe("UFCS")
+fn ufcsTests():
+  @it("should support method-style call")
+  fn shouldSupportMethodStyleCall():
     expect(double(5)).toEq(10)
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply tests/spec/functions.test.ry:102]: Skipping — this is a pre-existing test issue, not introduced by this PR.

This PR is a mechanical syntax migration (lambda `it("...", ():` → directive `@it("...") fn`). The body `expect(double(5)).toEq(10)` was already on line 98 of the lambda form before the migration (verified via `git show f5759042:tests/spec/functions.test.ry`). Fixing the test logic to actually exercise UFCS dispatch is a separate concern that should land independently of the migration; otherwise this 167-file mechanical PR risks scope creep and review noise. Filing as a follow-up issue is the correct path.

File: tests/spec/list_concat_arc.test.ry [hunk 2]
```diff
@@ -18,42 +22,42 @@ describe("list + list ARC retain (#1236)", ():
...
 
-  it("List<Map<str, int>> concat retains map values after source drop", ():
+  @it("List<Map<str, int>> concat retains map values after source drop")
+  fn listMapStrIntConcatRetainsMapValuesAfterSourceDrop():
     a: List<Map<str, int>> = [{"a": 1}, {"b": 2}]
     b: List<Map<str, int>> = [{"c": 3}]
     ys = a + b
     a = [{"z": 999}]
     b = [{"y": 888}]
     expect(len(ys)).toEq(3)
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply tests/spec/list_concat_arc.test.ry:42]: Skipping — this is a pre-existing test gap, not introduced by this PR.

The lambda form already only checked the outer length (`expect(ys).toHaveLen(3)` / `expect(len(ys)).toEq(3)` — verified via `git show f5759042:tests/spec/list_concat_arc.test.ry`). The migration was mechanical syntax conversion only. Strengthening the ARC retain-after-source-drop assertions belongs in a separate PR so the migration stays a pure syntax transformation. Filing as a follow-up issue.

File: tests/spec/net.test.ry [hunk 3]
```diff
@@ -21,24 +25,24 @@ describe("TCP socket API", ():
...
-  )
-  it("should return Err when connect to closed port", ():
+  @it("should return Err when connect to closed port")
+  fn shouldReturnErrWhenConnectToClosedPort():
     case connect("127.0.0.1", 19998):
       Ok(conn):
         close(conn)
         fail("expected Err but got Ok")
       Err(e):
         expect(true).toEq(true)
```
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply tests/spec/net.test.ry:43]: Skipping — this is a pre-existing test flakiness concern, not introduced by this PR.

The hard-coded ports `19991` / `19998` were already in the lambda form before the migration (verified via `git show f5759042:tests/spec/net.test.ry`). The migration was a mechanical syntax conversion. Switching to ephemeral-port reservation (`bind("127.0.0.1", 0)` → `listenerPort` → close → connect) is a real test-quality improvement, but it should land in a focused PR so reviewers can scrutinize the new pattern without it being buried in a 167-file syntax sweep. Filing as a follow-up issue.

**Comment by t0k0sh1** [comment]
@coderabbitai — review-summary nitpick handling

**1. `docs/reference/testing.md:37-65` — show `from testing import` before the first directive example: APPLIED**

Added `from testing import it, describe` at the top of the first three code blocks in the `Function-Based Syntax (recommended)` section (the introductory `@it` snippet, the grouped `@describe` snippet) and at the top of the `## Example` section's standalone snippet. Readers copy-pasting any of these now get a complete file. The smaller pedagogical sub-section illustrations (`#### Shared Setup`, `#### Nested @describe`, `mock`, `verify`) are intentionally left without imports because they are feature illustrations, not full-file templates — adding the import to every block would be noise.

**2. `tests/spec/filesystem.test.ry:22-66` — make filesystem tests self-contained: SKIPPED (out of scope)**

The cross-test ordering dependency (`/tmp/ry_fs_test/single_dir` reuse, missing setup in `shouldDetectADirectory`) was already present in the lambda form before this PR — verified via `git show f5759042:tests/spec/filesystem.test.ry`. This PR is a mechanical syntax migration (lambda `it("...", ():` → directive `@it("...") fn`); the test bodies were not touched. Hardening the filesystem test fixtures is a separate concern and should land in its own PR so the migration stays a pure syntax sweep. Filing as a follow-up issue.

File: tests/spec/functions.test.ry (hunk 1 above)
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply tests/spec/functions.test.ry:102]: Filed as #1604.

File: tests/spec/list_concat_arc.test.ry (hunk 2 above)
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply tests/spec/list_concat_arc.test.ry:42]: Filed as #1605.

File: tests/spec/net.test.ry (hunk 3 above)
[the remark this answers is not in the document]
Comment (t0k0sh1) [reply tests/spec/net.test.ry:43]: Filed as #1606.

**Comment by t0k0sh1** [comment]
Filing follow-ups (referenced in the previous comment):

- #1604 — `tests/spec/functions.test.ry`: exercise UFCS path in `shouldSupportMethodStyleCall`
- #1605 — `tests/spec/list_concat_arc.test.ry`: strengthen ARC retain-after-source-drop assertions
- #1606 — `tests/spec/net.test.ry`: use ephemeral port reservation instead of hard-coded ports
- #1607 — `tests/spec/filesystem.test.ry`: make per-test setup self-contained
